rkt/cli_apps.go

```diff
@@ -243,6 +244,68 @@ func (al *appsVolume) String() string {
 	return strings.Join(vs, " ")
 }
 
+// appMountVolume is for CRI style per-app-volumes
+// this is a mount and volume in a single argument
+// It is exactly like --volume, but with a "target" param
+type appMountVolume apps.Apps
+
+func (am *appMountVolume) Set(s string) error {
+	pairs, err := url.ParseQuery(strings.Replace(s, ",", "&", -1))
+	if err != nil {
+		return err
+	}
+
+	mount := schema.Mount{}
+
+	target, ok := pairs["target"]
+	if !ok {
+		return fmt.Errorf("need to pass target param")
+	}
+	if len(target) != 1 {
+		return fmt.Errorf("label %s with multiple values %q", "target", target)
+	}
+	mount.Path = target[0]
+
+	delete(pairs, "target")
+
+	vol, err := types.VolumeFromParams(pairs)
+	if err != nil {
+		return errwrap.Wrap(fmt.Errorf("error parsing volume component of MountVolume"), err)
+	}
+
+	mount.AppVolume = vol
+	mount.Volume = vol.Name
+
+	as := (*apps.Apps)(am)
+	if as.Count() == 0 {
+		return fmt.Errorf("an image is required before any MountVolumes")
+	} else {
```

Comment:
Technically true, but it's hard to parse these without an active app (especially if we want to expose this flag in `rkt run` eventually).